src/OrchardCore.Modules/OrchardCore.Media/Controllers/AdminController.cs

```diff
@@ -187,18 +196,14 @@ public async Task<ActionResult<object>> GetMediaItem(string path)
                 path = "";
             }
 
-            var section = _shellConfiguration.GetSection("OrchardCore.Media");
-
-            var allowedFileExtensions = section.GetSection("AllowedFileExtensions").Get<string[]>() ?? DefaultAllowedFileExtensions;
-
             var result = new List<object>();
 
             // Loop through each file in the request
             foreach (var file in files)
             {
                 // TODO: support clipboard
 
-                if (!allowedFileExtensions.Contains(Path.GetExtension(file.FileName), StringComparer.OrdinalIgnoreCase))
+                if (!_allowedFileExtensions.Contains(Path.GetExtension(file.FileName), StringComparer.OrdinalIgnoreCase))
```

Comment:
Use a `HashSet<string>(StringComparison.OrdinalIgnoreCase)` everywhere we store the extensions. (perf)

Reply:
Done